compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -131,15 +142,27 @@ object Completion {
     /**
      * Return the list of symbols that shoudl be included in completion results.
      *
-     * If the mode is `Import` and several symbols share the same name, the type symbols are
-     * preferred over term symbols.
+     * If several symbols share the same name, the type symbols appear before term symbols inside
+     * the same `Completion`.
+     */
+    def getCompletions(implicit ctx: Context): List[Completion] = {
+      val groupedSymbols = completions.toList.groupBy(_.name.stripModuleClassSuffix.toSimpleName).toList
+      groupedSymbols.map { case (name, symbols) =>
+        val typesFirst = symbols.sortWith((s, _) => s.isType)
```

Comment:
to be a valid sort, comparing two type symbols should return false, so I'd do `(s1, s2) => s1.isType && !s2.isType`